keyvalue/servers/src/main/java/tachyon/master/keyvalue/KeyValueMaster.java

```diff
@@ -270,6 +278,99 @@ private void createStoreInternal(long fileId) throws FileAlreadyExistsException
   }
 
   /**
+   * Deletes a completed key-value store.
+   *
+   * @param uri {@link TachyonURI} to the store
+   * @throws IOException if non-Tachyon error occurs
+   * @throws IsNotKeyValueStoreException if the uri exists but is not a key-value store
+   * @throws FileDoesNotExistException if the uri does not exist
+   * @throws TachyonException if other Tachyon error occurs
+   */
+  public synchronized void deleteStore(TachyonURI uri)
+      throws IOException, IsNotKeyValueStoreException, FileDoesNotExistException, TachyonException {
+    long fileId = mFileSystemMaster.getFileId(uri);
+    if (fileId == IdUtils.INVALID_FILE_ID) {
+      throw new FileDoesNotExistException(
+          String.format("Failed to deleteStore: path %s does not exist", uri));
+    }
+    if (!mCompleteStoreToPartitions.containsKey(fileId)) {
+      // TODO(cc): If it is an incomplete store, throw a better exception.
+      throw new IsNotKeyValueStoreException(
+          String.format("Failed to deleteStore: path %s is not a completed key-value store", uri));
+    }
+    mFileSystemMaster.deleteFile(uri, true);
+    deleteStoreInternal(fileId);
+    writeJournalEntry(newDeleteStoreEntry(fileId));
+    flushJournal();
+  }
+
+  // Internal implementation to deleteStore a key-value store.
+  private void deleteStoreInternal(long fileId) {
+    mCompleteStoreToPartitions.remove(fileId);
+  }
+
+  // Deletes a store, called when replaying journals.
+  private void deleteStoreFromEntry(DeleteStoreEntry entry) {
+    deleteStoreInternal(entry.getStoreId());
+  }
+
+  long getFileId(TachyonURI uri, String caller)
+      throws AccessControlException, FileDoesNotExistException {
+    long fileId = mFileSystemMaster.getFileId(uri);
+    if (fileId == IdUtils.INVALID_FILE_ID) {
+      throw new FileDoesNotExistException(
+          String.format("Failed to %s: path %s does not exist", caller, uri));
+    }
+    return fileId;
+  }
```

Comment:
Why is the handling different here (instead of being able to directly call `getFileId`)?